library/core/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1711,9 +1713,22 @@ public void initializeOutput(ExtractorOutput output, int trackId) throws ParserE
           nalUnitLengthFieldLength = hevcConfig.nalUnitLengthFieldLength;
           break;
         case CODEC_ID_FOURCC:
-          initializationData = parseFourCcVc1Private(new ParsableByteArray(codecPrivate));
-          if (initializationData != null) {
-            mimeType = MimeTypes.VIDEO_VC1;
+          initializationData = null;
+          Pair<Integer, List<byte[]>> pair = parseFourCcVc1Private(new ParsableByteArray(codecPrivate));
```

Comment:
Feels like things would be much simpler if the method returned a non-null Pair<String, List<byte[]>>, where the first part is the mimeType and can be set to `VIDEO_UNKNOWN`. You could then do something much more concise here, like:
```
Pair<String, List<byte[]>> mimeTypeAndInitializationData = ...
mimeType = mimeTypeAndInitializationData.first;
initializationData = mimeTypeAndInitializationData.second;
```
The `Log.w` call could be moved inside the method that's called here.